File: leetcode/python/11.盛最多水的容器.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        n = len(height)
        if n < 2 : return 0

        ret = 0
        i, j = 0, n - 1
        while i < j:
            if height[i] < height[j]:
                ret = max(ret, height[i] * (j - i))
                i += 1
            else:
                ret = max(ret, height[j] * (j - i))
                j -= 1
        return ret

File: leetcode/golang/11.盛最多水的容器.go
func maxArea(height []int) int {
    n := len(height)
    if n < 2 {
        return 0;
    }

    ret := 0;
    tmp := 0;
    i, j := 0, n - 1;
    for i < j {
        if height[i] < height[j] {
            tmp = height[i] * (j - i);
            i += 1;
        } else {
            tmp = height[j] * (j - i);
            j -= 1;
        }

        if tmp > ret {
            ret = tmp;
        }
    }
    return ret;
}